R/get_heartrate.R

```diff
@@ -222,10 +222,11 @@ get_hr_from_time_series <- function(x, sampling_rate, method = 'acf', min_hr = 4
     confidence <- NA
   }
   
+  
   # If hr or condidence is NaN, then return hr = 0 and confidence = 0
-  if (is.na(confidence) || is.na(hr)) {
-    confidence <- NA
-    hr <- NA
+  if ((length(hr) == 0) || is.null(hr)) {
```

Comment:
You check if `hr` is null here, but then reassign it to NA. Are these the same object or are you reusing a variable name? If `hr` is supposed to be non-numeric, then its non-value representation is `NULL`. If `hr` is numeric, its non-value representation is `NA`.

This is important because if you need to check for non-value-ness somewhere down the line, you should know whether it's numeric (check for NA) or otherwise (check for NULL).

Reply:
`hr` is a supposed to be a numeric value hence why it is recast into `NA`

As for the NULL check and length(hr)>0, they are there to ensure that the `hr` does not come out with weird results. For eg., if all the signal is 0, the output of L#193 is a list of `NaN'. And the heart rate estimated in the following 5 lines will give a numeric output, but a vector of length `0`. So the check. I also dropped in the NULL for safety.

So I don't believe in making any changes to these lines of code.

Reply:
Alright. I just find it confusing.